client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -134,6 +135,7 @@ type PlanFeatures2023GridType = PlanFeatures2023GridProps &
 
 type PlanFeatures2023GridState = {
 	showPlansComparisonGrid: boolean;
+	selectedStorage: { [ key: string ]: string | undefined };
```

Comment:
do we have enough definitions to create a more closed-form signature? something like:

```
type SelectedStorage = {
    [ key: string ]: ( typeof STORAGE_TERMS )[ number ] | null
};
```

It's still open on the keys, but that we check below anw. Ideally, `Map` ought to become the go-to type and structure really, as we can limit keys better I think.

------------------

Also, regarding `null` VS `undefined` - I tend to think of these as different. `undefined` for something that we have not observed yet, `null` for something that we have observed but set empty its value. So `null` like a go-to value for "known emptiness", and `undefined` akin to "unknown yet". I think Abramov would agree (in case you've done his Javascript Universe course) :D

^ not sure which to pick for this case. it sounds like `null` from the usage below, cause undefined would just result from key missing?

Reply:
Looking into this now 👀

Reply:
> do we have enough definitions to create a more closed-form signature

We have two different storage add-on tiers currently, but it might very well expand. Implemented the changes in [b254db7](https://github.com/Automattic/wp-calypso/pull/79041/commits/b254db7884ce6559e5c9d778037ec4cb30e928dd)